javav2/example_code/acmpca/src/main/java/com/example/acmpca/DeleteCertificateAuthority.java

```diff
@@ -0,0 +1,60 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.example.acmpca;
+
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.acmpca.AcmPcaClient;
+import software.amazon.awssdk.services.acmpca.model.AcmPcaException;
+import software.amazon.awssdk.services.acmpca.model.DeleteCertificateAuthorityRequest;
+
+// snippet-start:[acmpca.java2.DeleteCertificateAuthority.main]
+/**
+ * Before running this Java V2 code example, set up your development 
+ * environment, including your credentials.
+ * 
+ * For more information, see the following documentation topic:
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ */
+public class DeleteCertificateAuthority {
+  
+  public static void main(String[] args) throws Exception {
+
+    final String usage =
+        """
+            Usage: <region> <certArn> <restorePeriod>
+
+            Where:
+               region - the AWS region (e.g. us-east-1)
+               caArn - the ARN of the certificate authority to delete
+               restorePeriod - (optional) The number of days for restoration before permanent deletion occurs
+            """;
+
+    if (args.length < 2 || args.length > 3) {
+      System.out.println(usage);
+      return;
+    }
+
+    String region = args[0];
+    String caArn = args[1];
+    int restorePeriod = args.length == 3 ? Integer.parseInt(args[2]) : 30; 
```

Comment:
This should be `null` if the user didn't provide it. 30 is the default for the API, but we should be letting the API handle that if the user didn't provide anything instead of setting our own default parameter here.